src/routes/general.router.js

```diff
@@ -35,51 +37,53 @@ router.post(
 
   async (req, res, next) => {
     console.log('Started publish')
-    const accounts = await res.locals.ocean.accounts.list()
-    let result
-
-    // console.log(req.body.metadata);
-    const {
-      name,
-      author,
-      license,
-      files,
-      price,
-      type,
-      description,
-      copyrightHolder,
-      categories
-    } = req.body
-
-    try {
-      const newAsset = {
-        // OEP-08 Attributes
-        // https://github.com/oceanprotocol/OEPs/tree/master/8
-        main: {
-          ...AssetModel.main,
-          name,
-          dateCreated: new Date().toISOString().split('.')[0] + 'Z', // remove milliseconds
-          author,
-          license,
-          files,
-          price,
-          type
-        },
-        additionalAttributes: {
-          ...AssetModel.additionalAttributes,
-          description,
-          copyrightHolder,
-          categories: [categories]
+    if (res.locals.globaloptions.enablepublish) {
+      const accounts = await res.locals.ocean.accounts.list()
+      let result
+      const {
+        name,
+        author,
+        license,
+        files,
+        price,
+        type,
+        description,
+        copyrightHolder,
+        categories
+      } = req.body
+      try {
+        const newAsset = {
+          // OEP-08 Attributes
+          // https://github.com/oceanprotocol/OEPs/tree/master/8
+          main: {
+            type: type,
```

Comment:
this will actually expand correctly in js when variable name and key of object item are the same, that's why we had this in there before:

```suggestion
            type,
```

But works either way :-)